pr_agent/git_providers/bitbucket_server_provider.py

```diff
@@ -38,10 +38,31 @@ def __init__(
         self.diff_files = None
         self.bitbucket_pull_request_api_url = pr_url
         self.bearer_token = get_settings().get("BITBUCKET_SERVER.BEARER_TOKEN", None)
+        #Get username and password from settings
+        self.username = get_settings().get("BITBUCKET_SERVER.USERNAME", None)
+        self.password = get_settings().get("BITBUCKET_SERVER.PASSWORD", None)
```

Comment:
Username and password are only used locally; there’s no need to set them as instance attribute.